### PR TITLE
fix: type annotation for `num_of_dpus` in GlueJobHook

### LINKED ISSUE #29091: Incorrect type annotation for `num_of_dpus` in GlueJobOperator/GlueJobHook
### Apache Airflow Provider(s)

amazon

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==7.1.0

### Apache Airflow version

2.2.2

### Operating System

macOS Ventura 13.1

### Deployment

Docker-Compose

### Deployment details

_No response_

### What happened

When calling GlueJobOperator and passing 
`create_job_kwargs={"Command": {"Name": "pythonshell"}}` I need to specify MaxCapacity and based on the code [here](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/glue.py#L127) that's equal to _num_of_dpus_ and that parameter is integer as stated [here](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/glue.py#L68) 
Because I want to use pythonshell, AWS Glue offers to setup ranges between 0.00625 and 1 and that can't be achieved with integer. 
`When you specify a Python shell job (JobCommand.Name="pythonshell"), you can allocate either 0.0625 or 1 DPU. The default is 0.0625 DPU.`

I was trying to pass _MaxCapacity_ in `create_job_kwargs={"Command": {"Name": "pythonshell"}, "MaxCapacity": 0.0625}` but it throws the error. 

### What you think should happen instead

I think that parameter _num_of_dpus_ should be type double or MaxCapacity should be allowed to setup as double if pythonshell was selected in Command -> Name.

### How to reproduce

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by wonhyeongseo**
closes: https://github.com/apache/airflow/issues/29091

Simple fix of type annotation for `num_of_dpus` according to [AWS Glue Documentation](https://docs.aws.amazon.com/glue/latest/webapi/API_CreateJob.html#Glue-CreateJob-request-MaxCapacity)

@Taragolis may you please review this PR?

### CODE REVIEW

**Comment by Taragolis** [comment]
You also need to change in Operator (potentially in other places)

https://github.com/apache/airflow/blob/2e90c06c3eaf6a8cd0d7f4055274834158879440/airflow/providers/amazon/aws/operators/glue.py#L85

I still don't know what is better annotation for this case `float | None` or `int | float | None`

**Comment by Taragolis** [comment]
Yeah, statics check failed:

```console
airflow/providers/amazon/aws/hooks/glue.py:97: error: Incompatible types in
assignment (expression has type "float", variable has type "int")  [assignment]
                self.num_of_dpus = num_of_dpus
```

**Comment by wonhyeongseo** [comment]
Ok @Taragolis. Adding in `float` instead of replacing `int` for backward compatability. I checked the files for `num_of_dpus`, and the modified two files are all there is. May you please run the workflow again? Thanks!

**Review by Taragolis — APPROVED** [review]
pending approval (wait for CI)

**Comment by Taragolis** [comment]
🤣 `mypy` became crazy when tried to make static check against condition in Hook

https://github.com/apache/airflow/blob/c4c14c07a99b1554346a426aa947e68eb52ce31e/airflow/providers/amazon/aws/hooks/glue.py#L95

Add annotation here, something like: `self.num_of_dpus: int | float = 10`

**Comment by wonhyeongseo** [comment]
image
Updated as you suggested, @Taragolis! Thank you so much for guiding me. I hope to contribute more in the future. Have a great day/afternoon!

**Review by potiuk — APPROVED** [review]

**Pull request merged**
